### ISSUE TITLE
[BUG] XGBoost Demo n_categories > 2

### ISSUE THREAD

**Issue by pklemenkov** [rapidsai/notebooks#168]
**Is your feature request related to a problem? Please describe.**
`simulate_data` function accepts number of output classes and this number can be configured via `n_categories`. But XGBoost's objective is hardcoded to 'binary:logistic' which causes exception when `n_categories > 2`. 

**Describe the solution you'd like**
Remove `k=2` from `simulate_data` API to not confuse users and put it inside the function.

**Describe alternatives you've considered**
Check if `n_categories > 2` and change objective to `multi:softmax`.
